src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -98,19 +114,26 @@ private void checkHealth() {
     }
   }
 
-  public void stop() {
+  public synchronized void stop() {
```

Comment:
To clarify my previous question, would anything go wrong as of now (without this patch) if user calls `stop()` and `awaitShutdown()` multiple times? It seems fine to me because both `Test.stop()` and `Test.awaitShutdown()` (similarly for `Service`) should be idempotent and allow multiple calls.